homeassistant/components/amcrest/camera.py

```diff
@@ -43,12 +47,18 @@ def __init__(self, hass, amcrest):
         self._stream_source = amcrest.stream_source
         self._resolution = amcrest.resolution
         self._token = self._auth = amcrest.authentication
+        self._snapshot_lock = threading.Lock()
 
     def camera_image(self):
         """Return a still image response from the camera."""
-        # Send the request to snap a picture and return raw jpg data
-        response = self._camera.snapshot(channel=self._resolution)
-        return response.data
+        with self._snapshot_lock:
+            try:
+                # Send the request to snap a picture and return raw jpg data
+                return self._camera.snapshot(channel=self._resolution).data
+            except (RequestException, ReadTimeoutError, ValueError) as error:
```

Comment:
It's a shame the library does not handle these exceptions and raises its own instead.

Reply:
Good idea. I'm working on the library, too, so I can look at doing that. Thanks!